tests/spec/invalid_field_type.json

```diff
@@ -0,0 +1,8 @@
+[
+    {
+    "name": "string_field",
+    "doc": "doc",
+    "type": "invalidtype",
```

Comment:
I included this in the initial draft PR to show what the failing test UX is like:

```
E           jsonschema.exceptions.ValidationError: {'name': 'string_field', 'doc': 'doc', 'type': 'invalidtype', 'bytes': 255} is not valid under any of the given schemas
E           
E           Failed validating 'oneOf' in schema['items']:
E               {'oneOf': [{'allOf': [{'$ref': '#/$defs/RecapNull'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapBool'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapInt'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapFloat'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapString'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapBytes'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapList'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapMap'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapStruct'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapEnum'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]},
E                          {'allOf': [{'$ref': '#/$defs/RecapUnion'},
E                                     {'$ref': '#/$defs/RecapCommonFields'}]}]}
E           
E           On instance[0]:
E               {'bytes': 255,
E                'doc': 'doc',
E                'name': 'string_field',
E                'type': 'invalidtype'}

.venv/lib/python3.10/site-packages/jsonschema/validators.py:1302: ValidationError
==================================================================== short test summary info =====================================================================
FAILED tests/unit/meta_schema/test_json_schema.py::test_valid_schemas[tests/spec/invalid_field_type.json] - jsonschema.exceptions.ValidationError: {'name': 'string_field', 'doc': 'doc', 'type': 'invalidtype', 'bytes': 255} is not valid under any of the given schemas
```

Reply:
Updated the harness to find `.unimplemented.invalid.json` files and expect that they will throw as invalid schemas, but mark that test as expected to fail while unimplemented.